lib/ansible/modules/windows/win_toast.ps1

```diff
@@ -0,0 +1,97 @@
+   Fail-Json $result "Sorry, this version of windows, $osversion, does not support Toast notifications.  Toast notifications are available from version $lowest_version" 
+}
+
+$stopwatch = [system.diagnostics.stopwatch]::startNew()
+$now = [DateTime]::Now
+$default_title = "Notification: " + $now.ToShortTimeString()
+
+$params = Parse-Args $args -supports_check_mode $true
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+# $expire_at = Get-AnsibleParam -obj $params -name "msg" -type "str"
+$expire_seconds = Get-AnsibleParam -obj $params -name "expire_seconds" -type "int" -default 45
+$group = Get-AnsibleParam -obj $params -name "group" -type "str" -default "Powershell"
+$msg = Get-AnsibleParam -obj $params -name "msg" -type "str" -default "Hello world!"
+$popup = Get-AnsibleParam -obj $params -name "popup" -type "bool" -default $true
+$tag = Get-AnsibleParam -obj $params -name "tag" -type "str" -default "Ansible"
+$title = Get-AnsibleParam -obj $params -name "title" -type "str" -default $default_title
+
+$timespan = New-TimeSpan -Seconds $expire_seconds
+$expire_at = $now + $timespan
+$expire_at_utc = $($expire_at.ToUniversalTime()|Out-String).Trim()
+
+$result = @{
+    changed = $false
+    expire_seconds = $expire_seconds
+    expire_at = $expire_at.ToString()
+    expire_at_utc = $expire_at_utc
+    group = $group    
+    msg = $msg
+    popup = $popup
+    tag = $tag
+    title = $title
+    toast_sent = $false
+}
+
+# If no logged in users, there is no notifications service,
+# and no-one to read the message, so exit but do not fail
+# if there are no logged in users to notify.
+ 
+if ((get-process -name explorer -EA silentlyContinue).Count -gt 0){
+
+  [Windows.UI.Notifications.ToastNotificationManager, Windows.UI.Notifications, ContentType = WindowsRuntime] > $null
+  $template = [Windows.UI.Notifications.ToastNotificationManager]::GetTemplateContent([Windows.UI.Notifications.ToastTemplateType]::ToastText01)
+  
+  #Convert to .NET type for XML manipulation
+  $toastXml = [xml] $template.GetXml()
+  $toastXml.GetElementsByTagName("text").AppendChild($toastXml.CreateTextNode($title)) > $null
+  # TODO add subtitle
+  
+  #Convert back to WinRT type
+  $xml = New-Object Windows.Data.Xml.Dom.XmlDocument
+  $xml.LoadXml($toastXml.OuterXml)
+  
+  $toast = [Windows.UI.Notifications.ToastNotification]::new($xml)
+  $toast.Tag = $tag
+  $toast.Group = $group
+  $toast.ExpirationTime = $expire_at
+  $toast.SuppressPopup = -not $popup
+  
+  try {
+     $notifier = [Windows.UI.Notifications.ToastNotificationManager]::CreateToastNotifier($msg)
+     if (-not $check_mode) {
+        $notifier.Show($toast)
+        $result.toast_sent = $true
+        Start-Sleep -Seconds $expire_seconds
+     }
+  } catch {
+    $excep= $_
+    Fail-Json $result "Failed to create toast notifier, exception was $($excep.Exception.Message) stack trace $($excep.ScriptStackStrace)".
```

Comment:
A small one I just noticed, not a deal breaker. We now tend to do
```powershell
   Fail-Json -obj $result -message "Failed to create toast notifier, exception was $($excep.Exception.Message) stack trace $($excep.ScriptStackStrace)".
```
For bonus points you can add the stack trace to the standard result value `exception` meand for stack traces:
```powershell
   $result.exception = $excep.ScriptStackStrace)
   Fail-Json -obj $result -message "Failed to create toast notifier, exception was $($excep.Exception.Message)".
```
PS Yes, bad name, it should have been `traceback` or something similar.